libdnf/advisory/advisory_set_iterator.cpp

```diff
@@ -30,16 +30,14 @@ class AdvisorySetIterator::Impl : private libdnf::solv::SolvMap::iterator {
 private:
     Impl(const AdvisorySet & advisory_set)
         : libdnf::solv::SolvMap::iterator(advisory_set.p_impl->get_map()),
-          advisory_set{advisory_set} {}
-
-    Impl(const AdvisorySetIterator::Impl & advisory_set_iterator_impl) = default;
+          advisory_set{&advisory_set} {}
 
     AdvisorySetIterator::Impl & operator++() {
         libdnf::solv::SolvMap::iterator::operator++();
         return *this;
     }
 
-    const AdvisorySet & advisory_set;
+    const AdvisorySet * advisory_set;
```

Comment:
Did you need to make this change? I'm not entirely sure of the purpose of these iterator classes, a note to discuss this next year :slightly_smiling_face:

Reply:
This commit adds a copy assignment operator. But, C++ "reference" can only be initialized, not assigned.
Therefore, the object containing the reference can also only be initialized. ->That is the reason for the change.